### ISSUE TITLE
How to pass an argument to a job/suite

### ISSUE THREAD

**Issue by jan--f** [SUSE/teuthology#46]
For the DeepSea test suite ([see here](https://github.com/SUSE/ceph/tree/wip-deepsea-testing)) it would be handy to pass an argument to change the branch (and repo) DeepSea is pulled and installed from.

I understand that this could be achieved by through putting it into the yaml or append a yaml file. This file must live in the repo though. This would be the ceph repo, which makes it hard to test a certain DeepSea branch.

I wonder if I should add one or two new cli arguments to teuthology-suite (e.g. ```--deepsea-repo``` and ```--deepsea-branch```).
Another idea would be to add a more generic cli argument to supply, say a json string, that in turn is added to the config. For example ```--add-config='{deepsea: {repo: <repo_url>, branch: foo}}'```.

**Comment by smithfarm** [comment]
You can pass arbitrary yaml to `t-s` by putting it in a file and giving the filename as a positional parameter at the end of the command. See `teuthology-suite --help`.

**Comment by jan--f** [comment]
Yes, but this yaml file has to be tracked in the ceph repo. This makes ad-hoc testing of a certain branch rather inconvenient. I wonder if I could make this easier or if anything speaks against the addition of a new argument.

**Comment by smithfarm** [comment]
How is it inconvenient, exactly? You can override the values specified in the "official" yaml in the ceph repo.

**Comment by smithfarm** [comment]
See https://github.com/SUSE/ceph/blob/ses4/qa/distros/all/opensuse_42.2.yaml for an example.

**Comment by smithfarm** [comment]
By "inconvenient" you mean it's hard to make a file with a yaml snippet and then pass the filename on the t-o command line?

**Comment by jan--f** [comment]
Well as I understand it, I need to check it into the repo and push it into some remote repo, so the openstack instance (in our case) can pick it up. That is inconvenient compared to passing a cli argument.

**Comment by smithfarm** [comment]
Well, you need to have your deepsea test code in a remote repo, but certainly not the `overrides` yaml.

**Comment by smithfarm** [comment]
I did two experiments. First, I made a file `dummy.yaml` containing `ceph_cm: salt` and put it in my home directory on the laptop. Then I tried running `c-w ceph-qa-suite --suite dummy ~/dummy.yaml` (command abbreviated, you get the idea). This failed because the `~/dummy.yaml` file was note found on the teuthology VM.

Then I tried creating the file `/home/ubuntu/dummy.yaml` on the teuthology VM, and then I ran `c-w ceph-qa-suite --suite dummy /home/ubuntu/dummy.yaml` - this worked fine.
